daemon/watchdog_test.go

```diff
@@ -30,29 +30,30 @@ func must(err error) {
 func TestSdWatchdogEnabled(t *testing.T) {
 	mypid := strconv.Itoa(os.Getpid())
 	tests := []struct {
-		usec string // empty => unset
-		pid  string // empty => unset
+		usec     string // empty => unset
+		pid      string // empty => unset
+		unsetEnv bool
 
 		werr   bool
 		wdelay time.Duration
 	}{
 		// Success cases
-		{"100", mypid, false, 100 * time.Microsecond},
-		{"50", mypid, false, 50 * time.Microsecond},
-		{"1", mypid, false, 1 * time.Microsecond},
+		{"100", mypid, true, false, 100 * time.Microsecond},
+		{"50", mypid, true, false, 50 * time.Microsecond},
+		{"1", mypid, false, false, 1 * time.Microsecond},
+		{"1", "", true, false, 1 * time.Microsecond},
 
 		// No-op cases
-		{"", mypid, false, 0}, // WATCHDOG_USEC not set
-		{"1", "", false, 0},   // WATCHDOG_PID not set
-		{"1", "0", false, 0},  // WATCHDOG_PID doesn't match
-		{"", "", false, 0},    // Both not set
+		{"", mypid, true, false, 0}, // WATCHDOG_USEC not set
+		{"1", "0", false, false, 0}, // WATCHDOG_PID doesn't match
+		{"", "", true, false, 0},    // Both not set
 
 		// Failure cases
-		{"-1", mypid, true, 0},                // Negative USEC
-		{"string", "1", true, 0},              // Non-integer USEC value
-		{"1", "string", true, 0},              // Non-integer PID value
-		{"stringa", "stringb", true, 0},       // E v e r y t h i n g
-		{"-10239", "-eleventythree", true, 0}, // i s   w r o n g
+		{"-1", mypid, true, true, 0},                // Negative USEC
```

Comment:
Yes, just making sure that it works. Unless we have any specific corner-case we want to cover, in which case better perhaps to align them and introduce another tests group below.